### PR TITLE
Add linting to client code

### PR DESCRIPTION
**Pull request by jamesvclements**
Okay cool, so far this adds ESLint & Prettier to the client.

Basically it just consisted of adding an `.eslintrc` (because eslint is already part of CRA I didn't have to install it). Then I installed prettier & associated dependencies that allow it to be run in combination w/ eslint. I added an `npm lint` to the client that will run the linting. Not sure if this will get triggered by Travis or not, we may need to add a root `npm lint` that runs the nested ones.

Just an enhancement, but it would be cool to use [husky](https://medium.com/netscape/git-hooks-with-husky-8b98f2556363) to set up a git-hook that runs `tests` and `lints` on each `commit` or before `push`

Side note: I had to edit the `user settings` in VSCode to align w/ the Prettier rules I added to `.eslintrc` (just had to make it default to single quote) that way I can use the format shortcut to auto-format the code. You might need to do the same thing w/ <s>Atom</s> ST3. 

Setting this to blocked until #25 is merged as well as the linting for the server is added

### CODE REVIEW

**Comment by dpopp07** [comment]
Cool cool. I will follow your lead on adding that stuff and get it in the server side code. Once I do that, I'll push up and get your review and we can get this closed.

I'll see what I can do about getting Travis to lint the code. I should be able to set it up to fail if the code isn't linted properly, that's how our SDK repos are.

I've looked into using Husky before at work, it is pretty cool. I do think we should use it. I'll make an issue for that

**Comment by jamesvclements** [comment]
Depending on which gets merged first either this or #29 should add an `npm lint` to the root `package.json` that lints both

File: client/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (dpopp07) [inline client/.eslintrc]: From the eslint website:

> JavaScript - use .eslintrc.js and export an object containing your configuration.
YAML - use .eslintrc.yaml or .eslintrc.yml to define the configuration structure.
JSON - use .eslintrc.json to define the configuration structure. ESLint’s JSON files also allow JavaScript-style comments.
__Deprecated - use .eslintrc, which can be either JSON or YAML.__
package.json - create an eslintConfig property in your package.json file and define your configuration there.

You may want to change your config file to a .js file like the CRA app has, seems like .eslint files are deprecated.

Comment (jamesvclements) [reply client/.eslintrc]: Good catch

**Pull request merged**